logstash-core/lib/logstash/outputs/base.rb

```diff
@@ -56,34 +63,51 @@ def receive(event)
   def worker_setup
     if @workers == 1
       @worker_plugins = [self]
-      @worker_threads = []
     else
-      define_singleton_method(:handle, method(:handle_worker))
-      @worker_queue = SizedQueue.new(20)
+      define_singleton_method(:multi_handle, method(:handle_worker))
```

Comment:
I'll just add that it'd be nice to not do this sort of hackery but rather just use ivars for changing this stuff up, however that runs the risk of being a performance regression and is IMHO out of scope for this PR.